scripts/travisci/check_flake8.sh

```diff
@@ -78,44 +83,174 @@ E722 # do not use bare except, specify exception instead
+test_ignored_errors_all=(
+)
+
+# Files or directories to exclude from checks applied to all files.
+test_exclude_all=(
+)
+
+### CHANGED TEST FILES ###
+
+# Error codes applied to changed test files
+test_errors_changed=(
+${errors_changed[@]}
+)
+
+# Error codes ignored to changed test files
+test_ignored_errors_changed=(
+)
+
+# Files or directories to exclude from checks applied to changed test files.
+test_exclude_changed=(
+)
+
+### ADDED TEST FILES ###
+
+# Error codes applied to added test files
+test_errors_added=(
+${errors_changed[@]}
+)
+
+# Error codes ignored to added test files
+test_ignored_errors_added=(
 )
 
+# Files or directories to exclude from checks applied to added test files.
+test_exclude_added=(
+)
+
+
+################################################################################
+
 # DOCSTRING
 # The following error code enables all the error codes for docstring defined
 # here:
 # http://pep257.readthedocs.io/en/latest/error_codes.html
 # If one of the errors needs to be ignored, just add it to the ignore array.
-erros_add="${errors_change[@]} D"
+#erros_add="${errors_changed[@]} D"
 
-errors_change="${errors_change[@]}"
-ignored_errors_change="${ignored_errors_change[@]}"
 if [[ "${TRAVIS_PULL_REQUEST}" != "false" && "${TRAVIS}" == "true" ]]; then
   files_changed="$(git diff "${TRAVIS_COMMIT_RANGE}" --diff-filter=M \
-    --name-only)"
+                     --name-only | grep ".*\.py")"
   files_added="$(git diff "${TRAVIS_COMMIT_RANGE}" --diff-filter=A \
-    --name-only)"
-  flake8 --isolated --import-order-style=google \
-    --application-import-names=sandbox,garage,examples,contrib \
-    --select="${errors_change// /,}" \
-    --ignore="${ignored_errors_change// /,}" "${files_changed}"
-  flake8 --isolated --import-order-style=google \
-    --application-import-names=sandbox,garage,examples,contrib \
-    --select="${erros_add// /,}" \
-    --ignore="${ignored_errors_change// /,}" "${files_added}"
+                   --name-only | grep ".*\.py")"
 else
   git remote set-branches --add origin master
   git fetch
-  files_changed="$(git diff origin/master --diff-filter=M --name-only)"
-  files_added="$(git diff origin/master --diff-filter=A --name-only)"
-  flake8 --isolated --import-order-style=google \
-    --application-import-names=sandbox,garage,examples,contrib \
-    --select="${errors_change// /,}" \
-    --ignore="${ignored_errors_change// /,}" "${files_changed}"
-  flake8 --isolated --import-order-style=google \
-    --application-import-names=sandbox,garage,examples,contrib \
-    --select="${erros_add// /,}" \
-    --ignore="${ignored_errors_change// /,}" "${files_added}"
+  files_changed="$(git diff origin/master --diff-filter=M --name-only \
+                     | grep ".*\.py")"
+  files_added="$(git diff origin/master --diff-filter=A --name-only \
+                   | grep ".*\.py")"
+fi
+
+test_files_changed="$(echo "${files_changed}" |  grep "tests/*")"
```

Comment:
did `flake8 --exclude=tests ...` not work?

Reply:
Yes, the first three checks exclude "./tests/*", and the last three only check on "./tests/*". How can use --exclude in the last three checks that are only for "./tests*"? Some inverted regex?